lib/manager/poetry/artifacts.ts

```diff
@@ -107,7 +107,8 @@ function getSourceCredentialVars(
   const envVars: Record<string, string> = {};
 
   for (const source of poetrySources) {
-    const matchingHostRule = find({ url: source.url });
+    const matchingHostRule =
+      find({ hostType: 'pypi', url: source.url }) || find({ url: source.url });
```

Comment:
What's the benefit of using a hostType here? The fallback should also match all possible hostType (so it ignore it).

Reply:
The problem with this lookup is that there might be other host rules that match the URL but are meant for something different. This is especially problematic when you want to/need to consume packages from the `platform` for which an host-rule is created automatically.

Reply:
We do a very similar thing for [NuGet](https://github.com/renovatebot/renovate/blob/b0ce30b59a2c31a58b2f71972b89ad13832cde9f/lib/manager/nuget/artifacts.ts#L42
), [Docker](https://github.com/renovatebot/renovate/blob/93d32806b4145041579494968f7f4966d0514461/lib/datasource/docker/index.ts#L273) or [Packagist](https://github.com/renovatebot/renovate/blob/93d32806b4145041579494968f7f4966d0514461/lib/datasource/packagist/index.ts#L32) datasources.

Reply:
Ok so the PR description is misleading. You like to use a hostType to differentiate same hostname with different credentials in hostrules , so you don't need to have a lot more specific rules. 🤔

Can we update the PR description / issue requirements?
As in general we support passing credentials to poetry call.